standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/txn/TxnHandler.java

```diff
@@ -2557,7 +2583,7 @@ public void performWriteSetGC() throws MetaException {
     }
   }
 
-  protected long getMinOpenTxnIdWaterMark(Connection dbConn) throws MetaException, SQLException {
+  protected long getMinOpenTxnIdWaterMark(Connection dbConn) throws MetaException {
```

Comment:
What is `dbConn` used for? Can we achieve the same functionality with `jdbcTemplate` ?

Reply:
getMinOpenTxnIdWaterMark() is called from multiple places and `TxnHandler` cleanup is not part of this task. I made only the really necessary changes in it. Since place where this method is called in `TxnHandler` are still not using `jdbcTemplate`, therefore this method must still accept a `Connection` instance as parameter. With the cleanup of `TxnHandler` also, the signature can finally be changed and `jdbcTemplate` used instead.